src/ext/java/org/opentripplanner/ext/flex/FlexLocationsToStreetEdgesMapper.java

```diff
@@ -0,0 +1,68 @@
+package org.opentripplanner.ext.flex;
+
+import org.locationtech.jts.geom.Point;
+import org.opentripplanner.common.geometry.GeometryUtils;
+import org.opentripplanner.graph_builder.DataImportIssueStore;
+import org.opentripplanner.graph_builder.services.GraphBuilderModule;
+import org.opentripplanner.model.Location;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.core.TraverseModeSet;
+import org.opentripplanner.routing.edgetype.StreetEdge;
+import org.opentripplanner.routing.graph.GraphIndex;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.impl.StreetVertexIndex;
+import org.opentripplanner.routing.vertextype.StreetVertex;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class FlexLocationsToStreetEdgesMapper implements GraphBuilderModule {
+
+  @Override
+  public void buildGraph(
+      Graph graph, HashMap<Class<?>, Object> extra, DataImportIssueStore issueStore
+  ) {
+    if (graph.locationsById.isEmpty()) {
+      return;
+    }
+
+    StreetVertexIndex streetIndex = new StreetVertexIndex(graph);
+    TraverseModeSet carTraverseModeSet = new TraverseModeSet(TraverseMode.CAR);
+
+    for (Location location : graph.locationsById.values()) {
+      for (Vertex vertx : streetIndex.getVerticesForEnvelope(location
+          .getGeometry()
+          .getEnvelopeInternal())
+      ) {
+        if (!(vertx instanceof StreetVertex)) { continue; }
+        if (vertx
+            .getOutgoing()
+            .stream()
+            .noneMatch(edge -> edge instanceof StreetEdge && ((StreetEdge) edge).canTraverse(
```

Comment:
Add another check for walking

Reply:
Extract a business method to check if this can be used as a transfer point, and add to StreetEdge